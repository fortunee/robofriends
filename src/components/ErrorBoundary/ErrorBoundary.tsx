import React, { Component } from 'react';

interface EBState {
  hasError: boolean;
}

interface EBProps {}

class ErrorBoundary extends Component<EBProps, EBState> {
  constructor(props: EBProps) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch() {
    this.setState({ hasError: true });
  }

  render() {
    return (
      <>
        {this.state.hasError ? (
          <h1>Oops!!! Something went wrong please reload the page</h1>
        ) : (
          this.props.children
        )}
      </>
    );
  }
}

export default ErrorBoundary;
