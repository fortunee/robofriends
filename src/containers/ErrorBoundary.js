import React, { Component } from 'react';

class ErrorBoundary extends Component {
  constructor() {
    super();
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
  }

  render() {
    return (
      <>
        {this.state.hasError ? (
          <h1>Oops!!! Something went wrong please reload the page</h1>
        ) : (
          this.props.children
        )}
      </>
    );
  }
}

export default ErrorBoundary;
