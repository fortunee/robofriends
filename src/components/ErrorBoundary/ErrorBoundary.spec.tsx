import { shallow } from 'enzyme';
import React from 'react';

import ErrorBoundary from './ErrorBoundary';

describe('<ErrorBoundary />', () => {
  it('Should render a ErrorBoundary component', () => {
    const component = shallow(<ErrorBoundary />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('Should show error text when component has an error', () => {
    const ChildComponent = () => null;
    const component = shallow(
      <ErrorBoundary>
        <ChildComponent />
      </ErrorBoundary>
    );
    component.find(ChildComponent).simulateError(new Error());
    expect(component.state()).toEqual({ hasError: true });
    expect(component.text()).toEqual(
      'Oops!!! Something went wrong please reload the page'
    );
  });

  it('Should work as expected without errors', () => {
    const component = shallow(
      <ErrorBoundary>
        <h1>All good</h1>
      </ErrorBoundary>
    );
    expect(component.state()).toEqual({ hasError: false });
    expect(component.text()).toEqual('All good');
  });
});
