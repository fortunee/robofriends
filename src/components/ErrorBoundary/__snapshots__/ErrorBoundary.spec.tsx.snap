// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> Should render a ErrorBoundary component 1`] = `"<Fragment />"`;
